vic/drivers/shared_image/src/vic_init.c

```diff
@@ -792,6 +796,120 @@ vic_init(void)
 
         soil_moisture_from_water_table(&(soil_con[i]), options.Nlayer);
 
+        // Soil thermal node thicknesses and positions
```

Comment:
This section was originally in the old initialize_model_state() function, but it really should go here, as node spacing is completely determined from soil parameters and global param options.  This way, all soil_con values are populated by the time we exit vic_init().